### PR TITLE
Change to use single-key signature scheme in contracts

### PR DESCRIPTION
**Pull request by ureeves**
By updating the version of  `bls12_381_bls` to `0.4`, we can leverage the improved API to seamlessly change to the single-key signature scheme for both Moonlight and the stake contract, with minimal changes to the codebase.

The `consesus`, `node`, and `node-data` crates do go through some changes, in particular to use `BlsMultisigSignature` and `BlsMultisigPublicKey`. This results in a much more explicit usage of the multi-signature scheme than before.

Resolves: #2044

### CODE REVIEW

File: consensus/src/quorum/verifiers.rs [hunk 1]
```diff
@@ -131,18 +131,10 @@ pub fn verify_votes(
...
-
-        match pks.split_first() {
-            Some((first, rest)) => {
-                let mut apk = BlsAggPublicKey::from(first);
-                apk.aggregate(rest)?;
-                Ok(apk)
-            }
-            None => Err(StepSigError::EmptyApk),
-        }
+        Ok(BlsMultisigPublicKey::aggregate(&pks)?)
```
Comment (HDauven) [inline consensus/src/quorum/verifiers.rs:137]: Nice, I really like this API change. Didn't check the [PR on BLS-sign](https://github.com/dusk-network/bls12_381-bls/pull/19) before, but nice.

**Review by HDauven — DISMISSED** [review]
LGTM!

File: consensus/src/quorum/verifiers.rs [hunk 2]
```diff
@@ -131,18 +131,10 @@ pub fn verify_votes(
...
         let pks: Vec<_> =
             self.iter().map(|(pubkey, _)| *pubkey.inner()).collect();
-
-        match pks.split_first() {
-            Some((first, rest)) => {
-                let mut apk = BlsAggPublicKey::from(first);
-                apk.aggregate(rest)?;
-                Ok(apk)
-            }
-            None => Err(StepSigError::EmptyApk),
```
Comment (herr-seppia) [inline consensus/src/quorum/verifiers.rs:144]: Can you remove the EmptyApk variant?

Comment (ureeves) [reply consensus/src/quorum/verifiers.rs:144]: Yes I can!

Comment (ureeves) [reply consensus/src/quorum/verifiers.rs:144]: Done

**Review by herr-seppia — APPROVED** [review]
LGTM

File: node-data/src/ledger/faults.rs [hunk 3]
```diff
@@ -199,8 +200,8 @@ impl Fault {
         msg: &[u8],
     ) -> Result<(), BlsSigError> {
         let signature = sign_info.signature.inner();
-        let sig = BlsSignature::from_bytes(signature)?;
-        let pk = BlsAggPublicKey::from(sign_info.signer.inner());
+        let sig = BlsMultisigSignature::from_bytes(signature)?;
+        let pk = BlsMultisigPublicKey::aggregate(&[*sign_info.signer.inner()])?;
```
[the remark this answers is not in the document]
Comment (ureeves) [reply node-data/src/ledger/faults.rs:204]: I thought so as well, but didn't want to change in this PR

**Pull request merged**